features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/attachments/preview/AttachmentsPreviewView.kt

```diff
@@ -50,22 +48,22 @@ import me.saket.telephoto.zoomable.rememberZoomableState
 @Composable
 fun AttachmentsPreviewView(
     state: AttachmentsPreviewState,
-    onDismiss: () -> Unit,
     modifier: Modifier = Modifier,
 ) {
     fun postSendAttachment() {
         state.eventSink(AttachmentsPreviewEvents.SendAttachment)
     }
 
+    fun postCancel() {
+        state.eventSink(AttachmentsPreviewEvents.Cancel)
+    }
+
     fun postClearSendState() {
         state.eventSink(AttachmentsPreviewEvents.ClearSendState)
     }
 
-    if (state.sendActionState is SendActionState.Done) {
-        val latestOnDismiss by rememberUpdatedState(onDismiss)
-        LaunchedEffect(state.sendActionState) {
-            latestOnDismiss()
-        }
+    BackHandler(enabled = state.sendActionState !is SendActionState.Sending) {
```

Comment:
Do we really need this check?

Reply:
The cancellation during the sending state is emitting a `ClearSendState` event, so I did not want to change this behavior.